api/room/schema_query.py

```diff
@@ -308,6 +325,96 @@ def resolve_all_remote_rooms(self, info, return_all=None):
                         if not re.search(match, room.name)]
         return AllRemoteRooms(rooms=actual_rooms)
 
+    all_available_rooms = graphene.Field(
+        AllAvailableRooms,
+        start_date=graphene.String(required=True),
+        start_time=graphene.String(required=True),
+        end_date=graphene.String(required=True),
+        end_time=graphene.String(required=True),
+        time_zone=graphene.String(required=True),
+        description="Returns available rooms in a given period \
+            \n- start_time: Start time and date when you want to book room from\
+            [required]\n- end_time: time and date you want to book room upto\
+                [required]\n time_zone: The time zone of the location\
+                    [required]\n location: The location of the office's \
+                        room you want to book"
+    )
+
+    @Auth.user_roles('Admin', 'Super Admin')
+    def resolve_all_available_rooms(self, info, **kwargs):
+
+        time_zone = kwargs['time_zone']
+        empty_string_checker(time_zone)
+        validate_date_input(kwargs['start_date'], kwargs['end_date'])
+        validate_date_input(kwargs['start_time'], kwargs['end_time'])
+        format_range_dates(kwargs['start_date'], kwargs['end_date'])
+        format_range_time(kwargs['start_time'], kwargs['end_time'])
+
+        start_time = convert_date(kwargs['start_date'],
+                                  kwargs['start_time'],
+                                  time_zone
+                                  )
+        end_time = convert_date(kwargs['end_date'],
+                                kwargs['end_time'],
+                                time_zone
+                                )
+
+        """
+        list of all remote rooms in a user's location
+        """
+        locational_remote_rooms = Query.resolve_all_remote_rooms(
+            self, info, return_all=False)
+        all_rooms = []
+        for room in locational_remote_rooms.rooms:
+            all_rooms.append({
+                "id": room.calendar_id,
+                "name": room.name
+            })
+
+        """
+        list of all remote rooms and
+        for each room shows whether it is free or busy in a given period.
+        """
+        service = credentials.set_api_credentials()
+        all_calendars = get_google_api_calendar_list()['items']
+        all_remote_rooms = []
+        for room in all_calendars:
+            remote_rooms = {
+                "timeMin": start_time,
+                "timeMax": end_time,
+                "timeZone": time_zone,
+                "items": [
+                    {
+                        "id": room['id']
+                    }
+                ]
+            }
+            res = service.freebusy().query(body=remote_rooms).execute()
+            all_remote_rooms.append(res[u'calendars'])
+
+        """
```

Comment:
These are docstring comments and are not typically used to comment blocks/lines of code. Please refer to 👉 https://www.python.org/dev/peps/pep-0008/#documentation-strings